CommonMark/Parser/TableMethods.cs

```diff
@@ -0,0 +1,365 @@
+
+					if (c == '|' && (i == 0 || !Utilities.IsInlineSpanSymbol(asStr[i - 1])))
+					{
+						var text = sb.ToString();
+						Utilities.ClearStringBuilder(sb);
+
+						if (text.Length > 0)
+							MakeCell(text, row, ref offset);
+
+						offset += text.Length;
+
+						// skip the |
+						offset++;
+						continue;
+					}
+
+					if (c == '\\')
+					{
+						sb.Append(c);
+						if (i + 1 < asStr.Length)
+						{
+							if (Utilities.IsEscapableSymbol(asStr[i + 1]))
+								sb.Append(asStr[i + 1]);
+						}
+						i++;
+					}
+					else
+					{
+						sb.Append(c);
+					}
+				}
+			}
+
+			if (sb.Length > 0)
+			{
+				var text = sb.ToString();
+				Utilities.ClearStringBuilder(sb);
+
+				if (text.Length > 0)
+					MakeCell(text, row, ref offset);
+			}
+		}
+
+		private static void MakeCell(string text, Block row, ref int offset)
+		{
+			int length = text.Length;
+			string trimmedText = text.TrimStart();
+			var leadingWhiteSpace = length - trimmedText.Length;
+			trimmedText = trimmedText.TrimEnd();
+			var trailingWhiteSpace = length - leadingWhiteSpace - text.Length;
+
+			var cell = new Block(BlockTag.TableCell, row.SourcePosition + offset + leadingWhiteSpace);
+			cell.SourceLastPosition = cell.SourcePosition + trimmedText.Length;
+			cell.StringContent = new StringContent();
+			cell.StringContent.Append(trimmedText, 0, trimmedText.Length);
+
+			if (row.LastChild == null)
+			{
+				row.FirstChild = row.LastChild = cell;
+			}
+			else
+			{
+				row.LastChild.NextSibling = cell;
+				row.LastChild = cell;
+			}
+
+			cell.IsOpen = false;
+		}
+
+		static void MakeTableRows(Block table, StringBuilder sb)
+		{
+			var parts = table.StringContent.RetrieveParts();
+			var offset = 0;
+
+			for (var i = 0; i < parts.Array.Length; i++)
+			{
+				var line = parts.Array[i];
+				if (line.Length <= 0)
+					continue;
+
+				var lineLength = line.Length;
+				string actualLine = line.Source.Substring(line.StartIndex, line.Length);
+
+				// skip the header row
+				if (i != 1 && !string.IsNullOrEmpty(actualLine) && actualLine != " ")
+				{
+					var rowStartsInDocument = table.SourcePosition + offset;
+					var row = new Block(BlockTag.TableRow, rowStartsInDocument);
+					row.SourceLastPosition = rowStartsInDocument + lineLength;
+
+					row.StringContent = new StringContent();
+					row.StringContent.Append(actualLine, 0, actualLine.Length);
+
+					if (table.LastChild == null)
+					{
+						table.FirstChild = row;
+						table.LastChild = row;
+					}
+					else
+					{
+						table.LastChild.NextSibling = row;
+						table.LastChild = row;
+					}
+
+					MakeTableCells(row, sb);
+					row.IsOpen = false;
+				}
+
+				offset += lineLength;
+			}
+		}
+
+		internal static bool TryMakeTable(Block b, LineInfo line, CommonMarkSettings settings)
+		{
+			if ((settings.AdditionalFeatures & CommonMarkAdditionalFeatures.GithubStyleTables) == 0) return false;
+
+			var parts = b.StringContent.RetrieveParts().Array;
```

Comment:
the `RetrieveParts()` do not guarantee that they will be split on the line. A single part could contain multiple lines and also a single line could consist of multiple parts.